lib/ParseSIL/ParseSIL.cpp

```diff
@@ -6742,6 +6745,242 @@ bool SILParserTUState::parseSILDefaultWitnessTable(Parser &P) {
+        std::max_element(paramIndices.begin(), paramIndices.end());
+    indexSubset = IndexSubset::get(
+        P.Context, maxIndexRef ? *maxIndexRef + 1 : 0, paramIndices);
+    return false;
+  };
+  // Parse parameter and result indices.
+  IndexSubset *parameterIndices = nullptr;
+  IndexSubset *resultIndices = nullptr;
+  if (parseIndexSubset("parameters", parameterIndices))
+    return true;
+  if (parseIndexSubset("results", resultIndices))
+    return true;
+
+  // Parse a trailing 'where' clause (optional).
+  // This represents derivative generic signature requirements.
+  GenericSignature *derivativeGenSig = nullptr;
+  SourceLoc whereLoc;
+  SmallVector<RequirementRepr, 4> derivativeRequirementReprs;
+  if (P.Tok.is(tok::l_square) && P.peekToken().is(tok::kw_where)) {
+    P.consumeToken(tok::l_square);
+    bool firstTypeInComplete;
+    P.parseGenericWhereClause(whereLoc, derivativeRequirementReprs,
+                              firstTypeInComplete,
+                              /*AllowLayoutConstraints*/ false);
+    if (P.parseToken(tok::r_square, diag::sil_diff_witness_expected_keyword,
+                     "]"))
+      return true;
+  }
+
+  // Parse original function name.
+  SILFunction *originalFn;
+  if (parseFunctionName(originalFn))
+    return true;
+
+  // Resolve derivative requirements.
+  if (!derivativeRequirementReprs.empty()) {
+    SmallVector<Requirement, 4> requirements;
+    auto *whereClause = TrailingWhereClause::create(
+        originalFn->getModule().getASTContext(), whereLoc,
+        derivativeRequirementReprs);
+    convertRequirements(P, originalFn, whereClause->getRequirements(),
+                        requirements);
+    assert(requirements.size() == derivativeRequirementReprs.size());
+    derivativeGenSig = evaluateOrDefault(
+        P.Context.evaluator,
+        AbstractGenericSignatureRequest{
+            originalFn->getLoweredFunctionType()->getGenericSignature(),
+            /*addedGenericParams=*/{},
+            std::move(requirements)},
+            nullptr);
+  }
+
+  // Parse differentiability witness body.
+  SILFunction *jvp = nullptr;
+  SILFunction *vjp = nullptr;
+  if (P.Tok.is(tok::l_brace)) {
+    // Parse '{'.
+    SourceLoc lBraceLoc = P.Tok.getLoc();
+    P.consumeToken(tok::l_brace);
+    // Parse JVP (optional).
+    if (P.Tok.is(tok::identifier) && P.Tok.getText() == "jvp") {
+      P.consumeToken(tok::identifier);
+      if (P.parseToken(tok::colon, diag::sil_diff_witness_expected_keyword,
+                       ":"))
+        return true;
+      Scope Body(&P, ScopeKind::FunctionBody);
+      if (parseFunctionName(jvp))
+        return true;
+    }
+    // Parse VJP (optional).
+    if (P.Tok.is(tok::identifier) && P.Tok.getText() == "vjp") {
+      P.consumeToken(tok::identifier);
+      if (P.parseToken(tok::colon, diag::sil_diff_witness_expected_keyword,
+                       ":"))
+        return true;
+      Scope Body(&P, ScopeKind::FunctionBody);
```

Comment:
I'd suggest sticking to one naming convention throughout one piece of code.
```suggestion
      Scope body(&P, ScopeKind::FunctionBody);
```